### DISCUSSION TITLE
query something from yarnrc or json with curl only

### DISCUSSION THREAD

**Discussion by pheiduck** [curl/curl#8387] [Q&A] [answered]
I use this for query a electron Version (use **awk** and **cut**):
```
_electron=electron"$(curl -s "https://raw.githubusercontent.com/microsoft/vscode/main/.yarnrc" | awk '/target/{gsub(/("|",)/,"",$2);print $2}' | cut -d '.' -f 1)"
```
I use this for query vscode Version (use **python**):
```
pkgver="$(curl -s "https://raw.githubusercontent.com/microsoft/vscode/main/package.json" | python -c "import sys, json; print(json.load(sys.stdin)['version'])")"
```

Any way to Simplify this more and use **curl** only?

**Comment by jzakrzewski** [comment, marked as answer]
Arguably simpler ;) 
```
_electron=electron"$(curl -s "https://raw.githubusercontent.com/microsoft/vscode/main/.yarnrc" | sed -n -E 's/^target "([0-9]*)\..*"/\1/p')"
pkgver="$(curl -s "https://raw.githubusercontent.com/microsoft/vscode/main/package.json" | jq -r .version)"
```
But you cannot do this with the curl only because curl does not process the response content. It's not curl's job to do so.

**Reply by pheiduck** [reply]
For the electron part I'm with you, but for the pkgver part **jq** is not installed by default on most systems, so I leave it at **python**.
Thank you for the suggestions and explanation. :)
